### PR TITLE
chore: Set track_progress=True by default

### PR DESCRIPTION
**Pull request by matthewfeickert**
In the notebooks set `track_progress=True` so that when users run interactively or locally they get information on pseudo-experiments. However, to avoid having huge JupyterBook output (`track_progress=True` shows iterative updates of the progress bar in the JupyterBook output) have the CI switch this to `track_progress=False` during the build.

```
* Set `track_progress=True` for notebook files to give interactive users
  progress information.
* Set `track_progess=False` during the JupyterBook build in CI to avoid having
  the updates of the progress bar information frozen into the built book.
   - Use `git config --global --add safe.directory` to guard against GitHub Actions
     environment erroring with 'fatal: detected dubious ownership in repository'.
```

### CODE REVIEW

File: .github/workflows/deploy-jupyter-book.yml [hunk 1]
```diff
@@ -35,6 +35,10 @@ jobs:
       run: |
         python -m pip list
 
+    - name: Set track_progress=False for notebooks
+      run: |
+        grep --files-with-matches --recursive "track_progress=True" book | xargs sed --in-place 's/track_progress=True/track_progress=False/g'
```
Comment (matthewfeickert) [inline .github/workflows/deploy-jupyter-book.yml:40]:
```suggestion
        git config --global --add safe.directory /__w/pyhf-tutorial/pyhf-tutorial
        git grep --name-only "track_progress=True" book | xargs sed --in-place 's/track_progress=True/track_progress=False/g'
```

File: .github/workflows/deploy-jupyter-book.yml [hunk 2]
```diff
@@ -35,6 +35,11 @@ jobs:
       run: |
         python -m pip list
 
+    - name: Set track_progress=False for notebooks
+      run: |
+        git config --global --add safe.directory /__w/pyhf-tutorial/pyhf-tutorial
+        git grep --name-only "track_progress=True" book | xargs sed --in-place 's/track_progress=True/track_progress=False/g'
```
Comment (matthewfeickert) [inline .github/workflows/deploy-jupyter-book.yml:41]: Note that this doesn't do anything for the `Makefile` so that might not be ideal.

**Comment by matthewfeickert** [comment]
This is the build view from CI

image

File: .github/workflows/deploy-jupyter-book.yml (hunk 2 above)
Comment (matthewfeickert) [reply .github/workflows/deploy-jupyter-book.yml:41]: I asked about this in https://github.com/orgs/executablebooks/discussions/1067.

**Review by kratsg — APPROVED** [review]

**Pull request merged**